### PR TITLE
Modernize luigi client

### PR DESCRIPTION
**Pull request by maxmarkus**
Split up Luigi client to separate file per manager and refactored to classes.

### CODE REVIEW

File: docs/luigi-client-api.md [hunk 1]
```diff
@@ -209,6 +207,92 @@ LuigiClient.linkManager().goBack(true);
 
 ## uxManager
 
+**Extends LuigiClientBase**
```
Comment (jesusreal) [inline docs/luigi-client-api.md]: Looks like this part is now duplicated

File: client/src/luigi-client.js [hunk 2]
```diff
@@ -71,13 +67,13 @@ function luigiClientInit() {
...
 
-  function setAuthData(eventPayload) {
+  const setAuthData = eventPayload => {
     if (eventPayload) {
       authData = eventPayload;
     }
-  }
+  };
 
   window.addEventListener('message', function messageListener(e) {
```
Comment (jesusreal) [inline client/src/luigi-client.js]: As discussed, let´s have a unique event listener for `message` event ;)

File: client/src/luigi-client.js [hunk 3]
```diff
@@ -134,20 +113,20 @@ function luigiClientInit() {
...
 
 luigiClientInit();
 
-/** @namespace */
+/** @private */
 const LuigiClient = {
   /**
-   * Use the functions and parameters to define the lifecycle of listeners, navigation nodes, and Event data.
-   * @name lifecycle
+   * Use the functions and parameters to define the Lifecycle of listeners, navigation nodes, and Event data.
```
Comment (jesusreal) [inline client/src/luigi-client.js]: As discussed, let´s move the lifecycle methods to its own class ;)

File: client/src/luigi-client.js [hunk 4]
```diff
@@ -187,394 +166,46 @@ const LuigiClient = {
...
-        promises.alerts[settings.id].promise = new Promise(resolve => {
-          promises.alerts[settings.id].resolveFn = resolve;
-        });
-        return promises.alerts[settings.id].promise;
-      }
-    };
-  }
+  uxManager: () => uxManagerInstance
 };
 export default LuigiClient;
```
Comment (jesusreal) [inline client/src/luigi-client.js]: As discussed, we could try to make it a class and export here a unique instance

File: client/src/uxManager.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
+import { LuigiClientBase } from './baseClass';
+import { randomBytes } from 'crypto';
```
[the remark this answers is not in the document]
Comment (maxmarkus) [reply client/src/uxManager.js]: crypto is a standard package coming with NodeJs and does not need to be installed. https://nodejs.org/api/crypto.html

File: client/src/luigi-client.js (hunk 2 above)
Comment (maxmarkus) [reply client/src/luigi-client.js]: Will be done in separate PR for this ticket, for easier review.

File: client/src/luigi-client.js (hunk 3 above)
Comment (maxmarkus) [reply client/src/luigi-client.js]: done

File: client/src/luigi-client.js (hunk 4 above)
Comment (maxmarkus) [reply client/src/luigi-client.js]: done

File: client/src/luigi-client.js (hunk 2 above)
Comment (jesusreal) [reply client/src/luigi-client.js]: PR: https://github.com/SAP/luigi/pull/559

File: client/src/linkManager.js [hunk 6]
```diff
@@ -0,0 +1,220 @@
+import { LuigiClientBase } from './baseClass';
+
+/**
+ * The Link Manager allows you to navigate to another route. Use it instead of an internal router to:
+  - Route inside micro front-ends.
```
Comment (bszwarc) [inline client/src/linkManager.js]:
```suggestion
  - Provide routing inside micro front-ends.
```

**Pull request merged**